eng/SnippetGenerator/DirectoryProcessor.cs

```diff
@@ -37,35 +37,50 @@ public DirectoryProcessor(string directory)
 
         public void Process()
         {
-            foreach (var mdFile in Directory.EnumerateFiles(_directory, "*.md", SearchOption.AllDirectories))
-            {
-                Console.WriteLine($"Processing {mdFile}");
 
-                var text = File.ReadAllText(mdFile);
-                bool changed = false;
+            List<string> files = new List<string>();
+            files.AddRange(Directory.EnumerateFiles(_directory, "*.md", SearchOption.AllDirectories));
+            files.AddRange(Directory.EnumerateFiles(_directory, "*.cs", SearchOption.AllDirectories));
 
-                text = MarkdownProcessor.Process(text, s =>
+            foreach (var file in files)
+            {
+                string SnippetProvider(string s)
                 {
                     var selectedSnippets = _snippets.Value.Where(snip => snip.Name == s).ToArray();
                     if (selectedSnippets.Length > 1)
                     {
                         throw new InvalidOperationException($"Multiple snippets with the name '{s}' defined '{_directory}'");
                     }
+
                     if (selectedSnippets.Length == 0)
                     {
                         throw new InvalidOperationException($"Snippet '{s}' not found in directory '{_directory}'");
                     }
 
                     var selectedSnippet = selectedSnippets.Single();
-                    Console.WriteLine($"Replaced {selectedSnippet.Name}");
-                    changed = true;
+                    Console.WriteLine($"Replaced {selectedSnippet.Name} in {file}");
                     return FormatSnippet(selectedSnippet.Text);
-                });
+                }
+
+                var originalText = File.ReadAllText(file);
```

Comment:
We're going to dramatically increase the number of files that get scanned.  I wonder if it'd be more efficient to do it line by line?  Not something to fix now, but might be worth filing a tracking bug for.  (Doing a couple of perf measurements and speeding this up might make a good Tech Talk for the team?)